lib/mllp/receiver.ex

```diff
@@ -192,17 +193,16 @@ defmodule MLLP.Receiver do
       |> Map.merge(Keyword.get(opts, :transport_opts, %{}))
       |> update_transport_options(port)
 
-    allowed_clients =
-      Keyword.get(opts, :allowed_clients, [])
-      |> Enum.map(&normalize_ip/1)
-      |> Enum.reject(&is_nil(&1))
+    verify_peer = get_in(transport_opts, [:socket_opts, :verify]) == :verify_peer
 
+    allowed_clients = get_allowed_clients(verify_peer, opts)
     proto_mod = __MODULE__
 
     proto_opts = [
```

Comment:
This could be map (proto opts is declared as `any()` in ranch). Then we could just work with a map internally for beautiful and efficient pattern matching 😍  Thoughts?